### PR TITLE
[Rascal] Add `getConnections()` and `getConnectionDetails()` method types

### PR DESCRIPTION
**Pull request by jasonrowsell**
- `ConnectionDetails` interface type
- `Broker.getConnections()` method type: [link to original method](https://github.com/onebeyond/rascal/blob/9fb33da9c1f5323e94d22b7c26c3abf5c52dabf3/lib/amqp/Broker.js#L210)
- `BrokerAsPromised.getConnections()` method type: [link to original method](https://github.com/onebeyond/rascal/blob/9fb33da9c1f5323e94d22b7c26c3abf5c52dabf3/lib/amqp/BrokerAsPromised.js#L53)
- `Vhost.getConnectionDetails()` method type: [link to original method](https://github.com/onebeyond/rascal/blob/9fb33da9c1f5323e94d22b7c26c3abf5c52dabf3/lib/amqp/Vhost.js#L187)

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes
  - [Broker.getConnections](https://github.com/onebeyond/rascal/blob/9fb33da9c1f5323e94d22b7c26c3abf5c52dabf3/lib/amqp/Broker.js#L210)
  - [BrokerAsPromised.getConnections](https://github.com/onebeyond/rascal/blob/9fb33da9c1f5323e94d22b7c26c3abf5c52dabf3/lib/amqp/BrokerAsPromised.js#L53)
  - [Vhost.getConnectionDetails](https://github.com/onebeyond/rascal/blob/9fb33da9c1f5323e94d22b7c26c3abf5c52dabf3/lib/amqp/Vhost.js#L187)

### CODE REVIEW

File: types/rascal/index.d.ts [hunk 1]
```diff
@@ -165,6 +165,11 @@ export interface Recovery {
     immediateNack?: boolean | undefined;
 }
 
+export interface ConnectionInfo {
```
Comment (carpasse) [inline types/rascal/index.d.ts:168]: I would rename this to ConnectionDetails to match the [code](https://github.com/onebeyond/rascal/blob/master/lib/amqp/Vhost.js#L188)

**Review by carpasse — CHANGES_REQUESTED** [review]

**Review by carpasse — APPROVED** [review]

**Comment by jakebailey** [comment]
`rascal` is at v20 upstream, but the DT package here is only at v10. Are these new things from a new version of `rascal` such that the DT package here should be bumped to match?

**Comment by jakebailey** [comment]
Then the version should be bumped to 10.2 to match, at least.

**Review by jakebailey — CHANGES_REQUESTED** [review]
Marking as needing changes per the above

**Review by UlisesGascon — APPROVED** [review]
LGTM! Thanks for doing this!

Ping @cressie176 for awareness

**Review by jakebailey — APPROVED** [review]

**Comment by jasonrowsell** [comment]
Ready to merge

**Pull request merged**